spec.emu

```diff
@@ -412,11 +412,12 @@ location: https://github.com/tc39/proposal-measure/
       1. Let _fractionDigits_ be _options_.[[FractionDigits]].
       1. Let _significantDigits_ be _options_.[[SignificantDigits]].
       1. Let _value_ be _O_.[[Value]].
-      1. If _fractionDigits_ is not *undefined*, then
-        1. Set _value_ to RoundToFractionDigits(_value_, _fractionDigits_).
-      1. Else if _significantDigits_ is not *undefined*, then
-        1. Set _value_ to RoundToSignificantDigits(_value_, _significantDigits_).
+      1. If _fractionDigits_ is not *undefined*, set _value_ to RoundToFractionDigits(_value_, _fractionDigits_).
+      1. Else if _significantDigits_ is not *undefined*, set _value_ to RoundToSignificantDigits(_value_, _significantDigits_).
+      1. Otherwise, throw a *TypeError* exception.
       1. Let _N_ be OrdinaryObjectCreate(*"%Amount.prototype%"*, « [[Currency]], [[FractionDigits]], [[SignificantDigits]], [[Unit]], [[Value]] »).
+      1. If _unit_ is not *undefined* and _O_.[[Currency]] is not *undefined*, throw a *TypeError* exception.
+      1. If _currency_ is not *undefined* and _O_.[[Unit]] is not *undefined*, throw a *TypeError* exception.
```

Comment:
Thought, could leave fixing for later: We should allow using the same unit/currency as is already in the Amount.

Also, is it intentional to allow
```js
new Amount(42, { unit: 'meter' }).with({ unit: 'liter' })
```
as this would do?